executor/src/witgen/machines/fixed_lookup_machine.rs

```diff
@@ -269,6 +306,14 @@ impl<'a, T: FieldElement> FixedLookup<'a, T> {
             }
         };
 
+        // Update the multiplicities
+        *self
+            .multiplicities
+            .entry(identity_id)
+            .or_default()
+            .entry(row)
```

Comment:
In the end, we create a full vector of the full column and store it in memory. Doesn't it make sense to start with a vector (instead of a hashmap) to begin with?

Reply:
But, then I need to change this part